lib/puppet/provider/package/te_agent_bin.rb

```diff
@@ -14,16 +14,41 @@
   has_feature :versionable
   has_feature :install_options
 
+  confine :osfamily => [ :RedHat ]
+  confine :exists   => "/etc/init.d/twdaemon"
```

Comment:
Won't this prevent the provider from working if the agent isn't already installed?

Reply:
removed confines